toolvoice/scripts/inference/cascade_asr/asr_cascade_openai_confetti.py

```
@@ -0,0 +1,453 @@
+OPENAI_WS_URL = f"wss://api.openai.com/v1/realtime?model={OPENAI_REALTIME_MODEL}"
+HEADERS = (("Authorization", f"Bearer {os.environ.get('OPENAI_API_KEY', '')}"),)
+
+
+# -------------- I/O helpers --------------
+
+def read_json_from_file(path: str) -> Any:
+    with open(path, "r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+def parse_tool_cell(cell: str) -> Union[List[Any], Dict[str, Any], List[str]]:
```

Ruff (0.16.1)

[warning] Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)